site/app/templates/grading/electronic/Details.twig

```diff
@@ -697,5 +708,61 @@
         }
         location.reload();
     }
+
+    {# functions to handle regrading, they loop through sections in order to know if a confirm message should be displayed #}
+
+    $(function() {
+        $('.regrade-btn').click(function(e) {
+            let count = 0;
+            {% for section in sections %}
+                {% for row in section.rows %}
+                    {% if row.graded_gradeable.getAutoGradedGradeable().getActiveVersion() > 0 %}
+                       count+=1;
+                    {% endif %}
+                {% endfor %}
+            {% endfor %}
+            if(count > 50) {
```

Comment:
Take a look at what this creates.
![image](https://user-images.githubusercontent.com/55092742/122594984-c0b71480-d035-11eb-9ad7-7c5ce9d909a9.png)
This is a working solution but is going to add a lot of these calls. These calls should also not be needed to load the page (there could be hundreds). This feature is only designed to be used in rare cases so the performance of loading the page shouldn't drop for support of this. I think adding an Ajax call and having a backend route do the calculations would be a much better option here.